### ISSUE TITLE
Test test_column_stats.py::test_column_stats_string_column_minmax fails sporadically

### ISSUE THREAD

**Issue by G-D-Petrov** [man-group/ArcticDB#1111]
### Describe the bug

Test  `test_column_stats.py::test_column_stats_string_column_minmax` failed in [master build for Python 3.7](https://github.com/man-group/ArcticDB/actions/runs/7012936550/job/19078510932).

It failed with a segmentation fault and produced the following log:

```
test_column_stats.py::test_column_stats_string_column_minmax Fatal Python error: Segmentation fault

Thread 0x00007f0094c9d740 (most recent call first):
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/arcticdb/version_store/_store.py", line 832 in create_column_stats
  File "/tmp/parallel_test/1/tests/unit/arcticdb/test_column_stats.py", line 289 in test_column_stats_string_column_minmax
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/_pytest/python.py", line 204 in pytest_pyfunc_call
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/pluggy/_callers.py", line 64 in _multicall
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/pluggy/_manager.py", line 112 in _hookexec
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/pluggy/_hooks.py", line 433 in __call__
test_codec.py::test_round_trip_dim0[VariantCodec-data_type16-array16-shape16] PASSED [ 50%]
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/_pytest/runner.py", line 226 in call_and_report
  File "/opPASSED [  3%]t/python/cp37-cp37m/lib/python3.7/site-packages/_pytest/runner.py", line 133 in runtestprotocol
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/_pytest/runner.py", line 114 in pytest_runtest_protocol
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/pluggy/_callers.py", line 80 in _multicall
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/pluggy/_manager.py", line 112 in _hookexec
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/pluggy/_hooks.py", line 433 in __call__
test_codec.py::test_round_trip_dim0[VariantCodec-data_type17-array17-shape17] PASSED [ 50%]
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/_pytest/main.py", line 350 in pytest_runtestloop
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/pluggy/_callers.py", line 80 in _multicall
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/pluggy/_manager.py", line 112 in _hookexec
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/pluggy/_hooks.py", line 433 in __call__
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/_pytest/main.py", line 325 in _main
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/_pytest/main.py", line 271 in wrap_session
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/_pytest/main.py", line 318 in pytest_cmdline_main
test_codec.py::test_round_trip_dim0[VariantCodec-data_type18-array18-shape18] PASSED [ 50%]
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/pluggy/_callers.py", line 80 in _multicall
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/pluggy/_manager.py", line 112 in _hookexec
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/pluggy/_hooks.py", line 433 in __call__
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/_pytest/config/__init__.py", line 170 in main
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/_pytest/config/__init__.py", line 192 in console_main
  File "/opt/python/cp37-cp37m/lib/python3.7/site-packages/pytest/__main__.py", line 5 in <module>
  File "/opt/python/cp37-cp37m/lib/python3.7/runpy.py", line 85 in _run
test_column_stats.py::test_column_stats_duplicated_primary_index _code
  File "/opt/python/cp37-cp37m/lib/python3.7/runpy.py", line 193 in _run_module_as_main
```

### Steps/Code to Reproduce

``` python
from arcticdb import Arctic
import pandas as pd

df0 = pd.DataFrame(
    {"col_0": ["a", "b"], "col_1": [1, 2], "col_2": [6, 5]}, index=pd.date_range("2000-01-01", periods=2)
)
df1 = pd.DataFrame(
    {"col_0": ["c", "d"], "col_1": [3, 4], "col_2": [8, 7]}, index=pd.date_range("2000-01-03", periods=2)
)

def generate_symbol(lib, sym):
    lib.write(sym, df0)
    lib.append(sym, df1)
    expected_column_stats = lib.read_index(sym)
    expected_column_stats.drop(
        expected_column_stats.columns.difference(["start_index", "end_index"]),
        axis=1,
        inplace=True,
    )
    expected_column_stats = expected_column_stats.iloc[[0, 1]]
    expected_column_stats["v1.0_MIN(col_1)"] = [df0["col_1"].min(), df1["col_1"].min()]
    expected_column_stats["v1.0_MAX(col_1)"] = [df0["col_1"].max(), df1["col_1"].max()]
    expected_column_stats["v1.0_MIN(col_2)"] = [df0["col_2"].min(), df1["col_2"].min()]
    expected_column_stats["v1.0_MAX(col_2)"] = [df0["col_2"].max(), df1["col_2"].max()]
    return expected_column_stats


ac = Arctic("lmdb://test")
lib = ac.get_library("test", craete_if_missing=True)
generate_symbol(lib, sym)
column_stats_dict = {"col_0": {"MINMAX"}}
lib.create_column_stats(sym, column_stats_dict)
```

### Expected Results

No segmentation fault should be thrown

### OS, Python Version and ArcticDB Version

Python 3.7 | ArcticDB 4.1.0

### Backend storage used

LMDB

**Comment by alexowens90** [comment]
Suspect resolved by #1143, reopen if this turns out not to be the case

**Comment by G-D-Petrov** [comment]
Reopening as the test failed again here: https://github.com/man-group/ArcticDB/actions/runs/10420563796/job/28862343386
